src/router/index.ts

```diff
@@ -0,0 +1,422 @@
+  activeRouteStore.set(newRoute);
+  window.history.replaceState(newRoute, documentTitle, routeToPath(newRoute));
+}
+
+export const initialize = () => {
+  const { pathname, search, hash } = window.location;
+  replace(pathToRoute(pathname + search + hash));
+};
+
+export function pathToRoute(path: string | null): Route {
+  if (!path) {
+    return { type: "404" };
+  }
+
+  const url = new URL(path, "https://app.radicle.xyz");
+  // Pathname starts usually with a "/", we remove it to avoid bad interpretations
+  const segments = url.pathname.substring(1).split("/");
+
+  const type = segments.shift();
+  switch (type) {
+    case "registration": {
+      const nameOrDomain = segments.shift();
+      const activeView = segments.shift();
+      const owner = url.searchParams.get("owner");
+      const retry = url.searchParams.get("retry");
+
+      if (nameOrDomain) {
+        if (
+          activeView === "checkNameAvailability" ||
+          activeView === "register"
+        ) {
+          return {
+            type: "registration",
+            params: {
+              activeView: {
+                type: activeView,
+                params: { nameOrDomain, owner },
+              },
+            },
+          };
+        } else if (activeView === undefined) {
+          return {
+            type: "registration",
+            params: {
+              activeView: {
+                type: "view",
+                params: { nameOrDomain, retry: retry === "true" },
+              },
+            },
+          };
+        }
+      }
+      return {
+        type: "registration",
+        params: { activeView: { type: "validateName" } },
+      };
+    }
+    case "faucet": {
+      const view = segments.shift();
+      if (view === "withdraw") {
+        return {
+          type: "faucet",
+          params: {
+            activeView: {
+              type: "withdraw",
+              params: { amount: url.searchParams.get("amount") },
+            },
+          },
+        };
+      }
+      return { type: "faucet", params: { activeView: { type: "form" } } };
+    }
+    case "vesting":
+      return { type: "vesting" };
+    case "seeds": {
+      const host = segments.shift();
+      if (host) {
+        const urn = segments.shift();
+        if (urn) {
+          if (segments.length === 0) {
+            return {
+              type: "projects",
+              params: {
+                urn,
+                peer: null,
+                profile: null,
+                seed: host,
+                content: "tree",
+              },
+            };
+          }
+          const params = resolveProjectRoute(url, segments);
+          if (params) {
+            return {
+              type: "projects",
+              params: {
+                ...params,
+                search: url.search,
+                seed: host,
+                urn,
+              },
+            };
+          }
+          return { type: "404" };
```

Comment:
Can we make `404` be the default, instead of returning it everywhere?

Reply:
dc69cdae858cfdff0c53f7b50ad0b16edeb67afe fixes this, we return `null` as indicator that no route has been found, and make sure in the `initialize` function to return the `404` route if no route has been found